### PR TITLE
Disable `differing_test_runners` health check

### PR DESCRIPTION
**Pull request by sobolevn**
Added in https://github.com/HypothesisWorks/hypothesis/commit/cde4067be62afa9bcfa21214c6a0da0c5a882c90

Refs https://github.com/python/cpython/pull/108879

CC @Zac-HD

### CODE REVIEW

**Comment by AlexWaygood** [comment]
Should we maybe also pin the version of hypothesis we're installing in CI here? https://github.com/python/cpython/blob/main/.github/workflows/build.yml#L434

It seems fairly likely that Hypothesis could add other health checks in the future that would cause our CI to randomly start failing

**Comment by sobolevn** [comment]
@dependabot and @sobolevn will bump hypothesis from time to time 👍

File: Tools/requirements-tests.txt
Comment (AlexWaygood) [inline Tools/requirements-tests.txt:1]: Maybe `requirements-hypothesis.txt`? `Tools/requirements-tests.txt` feels too similar to the existing `Tools/requirements-dev.txt` file we already have (and I think it's unlikely we'd use the same file for test dependencies that aren't related to the Hypothesis CI job).

I'm not even sure the file belongs in the `Tools/` directory... maybe it should just be a top-level file?

Comment (sobolevn) [reply Tools/requirements-tests.txt:1]: Dependabot already knows about this dir and Hypothesis is clearly a tool :)
So, let's leave it there.

**Review by AlexWaygood — APPROVED** [review]
LGTM, thanks! Will wait a little bit in case Hugo/Zac/Ezio want to chime in, though :)

**Comment by Zac-HD** [comment]
I'm a big fan of pinning your _transitive_ dependencies too with e.g. the `pip-tools` package, so long as they're updated regularly - hypothesis itself has an automated weekly PR for that.

We most recently added a new health check in January 2021, but I'd pin anyway.  For example, a while ago Astropy's CI started failing because a new Hypothesis release was _better at finding numerical bugs_, and while you do want to find out about them (updates) you don't want them to break everyone's regression tests (pins).   See also [this blog post](https://blog.nelhage.com/post/two-kinds-of-testing/).

**Comment by sobolevn** [comment]
Two transitive deps that are installed are: `attrs` and `sortedcontainers`. I think we can skip pinning them (the first regression here can prove me wrong).

Thanks for your help 👍

**Review by hugovk — APPROVED** [review]

File: Tools/requirements-tests.txt
Comment (hugovk) [reply Tools/requirements-tests.txt:1]: We could move them all into some other directory later, maybe `.github/dependabot/`, but let's get the CI back to green first.

Comment (AlexWaygood) [reply Tools/requirements-tests.txt:1]: yeah, `Tools/` still doesn't feel ideal to me, but I don't feel strongly about where they should go 👍

**Pull request merged**